python/ray/data/_internal/logical/rules/zero_copy_map_fusion.py

```diff
@@ -0,0 +1,79 @@
+from abc import abstractmethod
+from typing import List
+
+from ray.data._internal.execution.operators.map_operator import MapOperator
+from ray.data._internal.execution.operators.map_transformer import (
+    BlockMapTransformFn,
+    BuildOutputBlocksMapTransformFn,
+    MapTransformFn,
+    MapTransformFnDataType,
+)
+from ray.data._internal.logical.interfaces.optimizer import Rule
+from ray.data._internal.logical.interfaces.physical_plan import PhysicalPlan
+
+
+class ZeroCopyMapFusionRule(Rule):
+    """Base class for zero-copy map fusion rules.
+
+    Subclasses implement the optimization strategies for different combinations of
+    fused map operators, by dropping unnecessary data conversion `MapTransformFn`s.
```

Comment:
Add more to the docstring? What is this rule doing? When should subclasses override this rule?